common/test_rescue.yml

```diff
@@ -58,6 +58,19 @@
         download_file_fail_ignore: true
       when: vm_dir_name is defined and vm_dir_name
 
+    - include_tasks: esxi_download_datastore_file.yml
+      vars:
+        src_datastore: "{{ datastore }}"
+        src_file_path: "{{ vm_dir_name }}/{{ vm_serial_port_output_file | basename }}"
+        dest_file_path: "{{ current_test_log_folder }}/{{ vm_serial_port_output_file | basename }}"
+        download_file_fail_ignore: true
+      when: 
+        - vm_dir_name is defined
+        - vm_dir_name | bool
+        - vm_serial_port_output_file is defined
+        - vm_serial_port_output_file | bool
```

Comment:
Why do you convert string to a boolean? Without convert, it will check whether vm_dir_name or vm_serial_port_output_file is empty string, I think it will be enough. No need to convert to boolean.

Reply:
Done. Thanks